src/index.ts

```diff
@@ -0,0 +1,376 @@
+// We need to import the augmented definitions "somewhere" in our project, however since we have
+// it in tsconfig as an override and the api/types has imports, it is not strictly required here.
+// Because of the tsconfig override, we could import from '@polkadot/{api, types}/augment'
+import './interfaces/augment-api';
+import './interfaces/augment-types';
+// import '@polkadot/api/augment';
+// import '@polkadot/types/augment';
+import protoType from '@fragcolor/protos';
+
+// all type stuff, the only one we are using here
+// import {AccountsInfo} from "sample-polkadotjs-typegen/interfaces/protos";
+
+// external imports
+import { ApiPromise, Keyring } from '@polkadot/api';
+
+// our local stuff
+import * as definitions from './interfaces/definitions';
+import { ProtosCategories } from "@polkadot/types/lookup";
+import { Categories, GetGenealogyParams, GetProtosParams } from './interfaces/protos';
+import { GetDefinitionsParams, GetInstanceOwnerParams, GetInstancesParams } from './interfaces/fragments';
+import { u16, Vec } from '@polkadot/types-codec';
+
+
+export type availableCategories = {
+    text?: 'plain' | 'json',
+    trait?: Vec<u16>, // or should this be Array<number>??
+    shards?: 'edn' | 'binary',
+    audio?: 'oggFile' | 'mp3File',
+    texture?: 'pngFile' | 'jpgFile',
+    vector?: 'svgFile' | 'ttfFile',
+    video?: 'mkvFile' | 'mp4File',
+    model?: 'gltfFile' | 'sdf' | 'physicsCollider',
+    binary?: 'wasmProgram' | 'wasmReactor' | 'blendFile',
+};
+
+export type getProtosFuncParams = {
+    desc: boolean, 
+    fromIndex: number, 
+    limit: number, 
+    metadatKeys: Array<string>, 
+    categories: Array<availableCategories>, 
+    available: boolean|null, 
+    tags: Array<string>, 
+    returnOwers: boolean, 
+    owner: string|null, 
+    excludeTags: Array<string>
+}
+
+export type protoUploadFuncParams = {
+    references: Array<string>, 
+    category: availableCategories, 
+    tags: Array<string>,
+    linkedAssets: string|null, 
+    license: string, 
+    data: string
+}
+
+export type protoSetMetadataFuncParams = {
+    protoHash: string, 
+    metadataKey: string, 
+    data: string
+}
+
+export type getProtosGenealogyFuncParams = {
+    getAncestor: boolean, 
+    protoHash: string
+}
+
+export type fragmentsGetDefinitionsFuncParams = {
+    desc: boolean, 
+    fromIndex: number, 
+    limit: number, 
+    returnOwners?: boolean
+}
+
+export type fragmentsGetInstancesFuncParams = {
+    desc: boolean, 
+    fromIndex: number, 
+    limit: number,
+    definitionHash: string
+}
+
+export type fragmentsGetInstanceOwnerFuncParams = {
+    definitionHash: string, 
+    editionId: number, 
+    copyId: number
+}
+
+export async function protoUpload(protoUploadParams: protoUploadFuncParams): Promise<any> {
```

Comment:
@paulinechi - **for all the extrinsics** - let's just define the object in the brackets rather than creating a new type for the object. 

So Instead of creating the new type `protoUploadFuncParams`, let's just define the object in the brackets like this:

```javscript
protoUpload({
    references: Array<string>, 
    category: availableCategories, 
    tags: Array<string>,
    linkedAssets: string|null, 
    license: string, 
    data: string
})
```

Reply:
^

Reply:
> 

protoUpload({}) is not correct syntax, we can either change it to
export async function protoUpload(protoUploadParams: {
    references: Array<string>, 
    category: availableCategories, 
    tags: Array<string>,
    linkedAssets: string|null, 
    license: string, 
    data: string
}): Promise<any> 

or 

export async function protoUpload(
    references: Array<string>, 
    category: availableCategories, 
    tags: Array<string>,
    linkedAssets: string|null, 
    license: string, 
    data: string
): Promise<any> 

Is there any reason why you think creating a new type is not easy to understand?

Reply:
actually never mind. instead of defining the type in the parameter - let's just do it this way for all extrinsics and even RPCs (**since this way allows us to give certain parameters default values** and it is also easy for the programmer to read):

```typescript
export async function protoUpload({
references = [],
category,
tags = [],
linkedAssets = null,
license,
data
}: protoUploadFuncParams): Promise
```

Reply:
> actually never mind. instead of defining the type in the parameter - let's just do it this way for all extrinsics and even RPCs (**since this way allows us to give certain parameters default values** and it is also easy for the programmer to read):
> 
> ```ts
> export async function protoUpload({
> references = [],
> category,
> tags = [],
> linkedAssets = null,
> license,
> data
> }: protoUploadFuncParams): Promise
> ```

actually nevermind. what you've done is fine. with examples, it will be easy for a developer to understand.